### PR TITLE
Issue #403: Malware notifications

### PR DESCRIPTION
**Pull request by DonNateR**

### CODE REVIEW

**Comment by jonathan-buttner** [comment]
I'm not as familiar with the malware notification functionality, @kevinlog is this related to the policy work?

**Comment by jmikell821** [comment]
@ferullo can you please review this PR? I couldn't bring your name up in the list of assigned reviewers for some reason.

**Review by ferullo — APPROVED** [review]

File: docs/getting-started/install-endpoint.asciidoc [hunk 1]
```diff
@@ -107,6 +107,19 @@ After you have installed the agent, malware prevention is automatically enabled
 [role="screenshot"]
 image::images/install-endpoint/malware-protection.png[]
 
+[discrete]
+[[malware-notifications]]
+== Malware notications
+
+Platinum and enterprise users can customize the notifications on their host devices. These notifications appear whenever malware is detected or prevented with the syntax `Elastic Security {action} {filename}`.
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:114]:
```suggestion
Notifications appear on protected hosts whenever malware is detected or prevented. Platinum and enterprise users can customize these notifications using the `Elastic Security {action} {filename}`syntax. 
```

File: docs/getting-started/install-endpoint.asciidoc [hunk 2]
```diff
@@ -107,6 +107,19 @@ After you have installed the agent, malware prevention is automatically enabled
...
+
+Policy actions determine default notification behaviors:
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:116]:
```suggestion
Policy configuration determines default notification behavior: 
```

File: docs/getting-started/install-endpoint.asciidoc [hunk 3]
```diff
@@ -107,6 +107,19 @@ After you have installed the agent, malware prevention is automatically enabled
...
 
+[discrete]
+[[malware-notifications]]
+== Malware notications
+
+Platinum and enterprise users can customize the notifications on their host devices. These notifications appear whenever malware is detected or prevented with the syntax `Elastic Security {action} {filename}`.
+
+Policy actions determine default notification behaviors: 
+
+- **Prevent**: Notifications appear by default. Deselect the **Notify User** option to hide them.
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:118]:
```suggestion
- **Prevent**: Notifications appear by default. Deselect the **Notify User** option to disable them.
```

File: docs/getting-started/install-endpoint.asciidoc [hunk 4]
```diff
@@ -107,6 +107,19 @@ After you have installed the agent, malware prevention is automatically enabled
...
+- **Detect**: Notifications do not appear. Select the **Notify User** option to begin receiving them.
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:119]:
```suggestion
- **Detect**: Notifications do not appear. Select the **Notify User** option to enable them.
```

File: docs/getting-started/install-endpoint.asciidoc [hunk 5]
```diff
@@ -107,6 +107,19 @@ After you have installed the agent, malware prevention is automatically enabled
...
+
+If the malware policy is not selected, notifications will not appear on the host device.
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:121]:
```suggestion
If malware protection is disabled, notifications will not appear on the host device.
```

**Review by jmikell821 — CHANGES_REQUESTED** [review]

**Comment by jmikell821** [comment]
@DonNateR - a few comments. Make this a priority to get these changes in ASAP because I need to tag the preview -- we may have another edit from the endpoint team based on a bug that was discovered. 

1) Please put up a [preview link](https://security-docs_465.docs-preview.app.elstc.co/guide/en/security/master/install-endpoint.html) in each PR. Looking at this in the UI, this small section looks disjointed from the rest of the config paragraph above. 
2) If you look at the latest Kibana UI, it's changed to include the malware notifications. Your current screenshot is outdated. You need to update the "Malware Settings" paragraph. Change the title to "**Configure an Integration policy**." (I'm guessing when calling it by its full name, `Integration policy`, it's sentence case. If just referring to its short `Policy` name, capitalize, for now. See a panoramic capture I've attached -- slice and dice as appropriate. 
3) Mention that events for Windows, macOS, and Linux are enabled by default, but the customer can deselect if they want to turn these off.  
4) Move the piece about malware notifications with the rest of the "configure policy" bit above b/c it does not flow sitting at the bottom here. You also need to fix the typo in this title. If you put this above where it belongs, you can probably omit this title and make a subheading like "Configure malware protection" or something. 
5) In general, make sure you run a spell check before publishing a PR. 
6) [PR 451](https://github.com/elastic/security-docs/pull/451/files) folds into this topic. Keep that in mind when merging. 

image

File: docs/getting-started/install-endpoint.asciidoc [hunk 6]
```diff
@@ -94,27 +94,35 @@ If the prompt does not appear because you're using a version before macOS Big Su
...
 If you download the Elastic Agent on Windows 10 or above, you can configure Elastic Security as your antivirus software by doing the following:
 
 1. Go to **Fleet** > **Agents**. A list of hosts with the Elastic Agent appears. 
 2. Select your desired host from the list.
 3. From the host's page, find your named **Endpoint** integration and select it. 
-4. On the Integration Edit page, look for the **Protections** section and find **Type: Register as antivirus**. Toggle this option to enable. 
+4. On the Integration Edit page, look for the **Settings** section and find **Type: Register as antivirus**. Toggle this option to enable. 
 +
 [role="screenshot"]
 image::images/register-as-antivirus.png[]
```
Comment (jmikell821) [inline docs/getting-started/install-endpoint.asciidoc:128]: I'm a little lost here. Why do they need to do all this when they can select the option at the bottom? I can understand if they're in the Fleet app, but if they are following this page, they'll already be on the Policy config page. 

image

**Review by jmikell821 — CHANGES_REQUESTED** [review]

File: docs/getting-started/install-endpoint.asciidoc (hunk 6 above)
Comment (DonNateR) [reply docs/getting-started/install-endpoint.asciidoc:128]: These were written in a separate PR initially, but I was writing the content I discovered that this should be grouped together. Consolidating these in my next commit.

**Comment by jmikell821** [comment]
@DonNateR parts of this still look disconnected. You have users saving policy config settings twice --which is unnecessary. The "Register as Antivirus" option is on the same page and should be a part of the other sections. You are still writing this like it's separate from the Policy. Yes, it's optional -- but there still needs to be a logical flow. 

You need to check your numbering because you stop at 2 and list the rest of the steps (e.g, "Click Save to save changes to the Policy".) You may need to add a continuation character `+` in between each line break to ensure the ordered numbering stays intact. 
image

image

I also asked you review this section about verifying endpoint enrollment. You mention the status message but that's it. They can _click_ this link to view the progress and/or they can check "Agent Status" column to see if it's online. 
image

Please make sure all my comments are addressed and most importantly, triple check the preview UI and check your work to make sure it is error-free as possible.

**Comment by DonNateR** [comment]
[Malware notification updates](https://security-docs_465.docs-preview.app.elstc.co/guide/en/security/master/install-endpoint.html)

**Comment by jmikell821** [comment]
> If the prompt does not appear because you’re using a version before macOS Big Sur (11.0), enable the extension by doing the following: 

Change `before` to `earlier than`. 


> Confirm the kernel extension has loaded, enter kextstat | grep co.elastic.kendpoint.

Enter kextstat....etc to confirm the kernel extension has loaded. 

> To access the security integration policy:

Capitalize `Integration` to keep it consistent. 

> On the Integration Edit page, look for the Settings section and find Type: Register as antivirus. Toggle this option to enable.

Change `look for` to `go to`

> After you have customized your desired policy settings, click Save.

Capitalize Policy (for now). 

> After installing the Elastic Agent, there’s a lag time of several hours between when the Elastic Endpoint begins detecting and sending alerts to {Kibana}. 

The attribute has brackets around it in the UI. Make sure it's `{kib}` in your editor. Also, I think the steps to verify enrollment would look cleaner as a numbered procedure.

**Pull request merged**